src/commands/test/lib/junit-xml-util.ts

```diff
@@ -12,46 +11,26 @@ export class JUnitXmlUtil extends XmlUtil {
     let mainXml: Document = this.getEmptyXmlDocument();
 
     const self: JUnitXmlUtil = this;
-    return new Promise<Document>((resolve, reject) => {
-      fs.createReadStream(pathToArchive)
-        .pipe(unzip.Parse())
-        .on("entry", function (entry: unzip.Entry) {
-          // Skip directories and hidden system files
-          if (entry.type === "Directory" || path.basename(entry.path).substring(0, 1) === ".") {
-            return;
-          }
-          const fullPath: string = path.join(tempPath, entry.path);
-          entry.pipe(fs.createWriteStream(fullPath).on("close", () => {
-            try {
-              // Handle DOMParser warnings, errors and fatalErrors like JS exceptions
-              const configuration: object = {
-                locator: {},
-                errorHandler: function (level: string, msg: string) {
-                  throw `DOMParser${level}: ${msg}`;
-                }
-              };
-
-              const xml: Document = new DOMParser(configuration).parseFromString(fs.readFileSync(fullPath, "utf-8"), "text/xml");
-
-              let name: string = "unknown";
-              const matches: RegExpMatchArray = entry.path.match("^(.*)_TEST.*");
-              if (matches && matches.length > 1) {
-                name = matches[1].replace(/\./gi, "_");
-              }
-
-              self.appendToTestNameTransformation(xml, name);
-              self.removeIgnoredTransformation(xml);
-
-              mainXml = self.combine(mainXml, xml);
-            } catch (e) {
-              reject(e);
-            }
-          }));
-        })
-        .on("close", () => {
-          resolve(mainXml);
-        });
-    });
+
+    return this.getMergeXmlResultsPromise(pathToArchive, tempPath,
+      (fullPath: string, entry: unzip.Entry) => {
+        const xml: Document = new DOMParser(XmlUtil.DOMParserConfig).parseFromString(fs.readFileSync(fullPath, "utf-8"), "text/xml");
+
+        let name: string = "unknown";
+        const matches: RegExpMatchArray = entry.path.match("^(.*)_TEST.*");
+        if (matches && matches.length > 1) {
+          name = matches[1].replace(/\./gi, "_");
+        }
+
+        self.appendToTestNameTransformation(xml, name);
+        self.removeIgnoredTransformation(xml);
```

Comment:
We should investigate if there is there any reason why we can't add this:`self.removeEmptySuitesTransformation(xml);`? I think if we're smart about it we may be able to refactor this even further so that lines 17 to 28 here can be shared by both JUnit and NUnit classes? (Please correct me it I'm mistaken!)

Reply:
NUnit and JUnit use different xml structure. E.g. `removeIgnoredTransformation` are completely different. I see that we have empty test-suite only inside nunit xml.

Reply:
That's a shame - could we do something like this? (please excuse the pseudo-code):

```
get the xml document (shared code)
perform regex matching (shared code)
process xml (these `self.appendToTestNameTransformation`, `self.removeIgnoredTransformation` etc) (class specific override of abstract function)
combine xml (shared code)
```